### ISSUE TITLE
No order of items in content selectors

### ISSUE THREAD

**Issue by Holzhauer** [JabRef/jabref#3791]
- [x] I have tested the latest master version from http://builds.jabref.org/master/ and the problem persists

JabRef version 4.1

Steps to reproduce:

1. Add a number of entries to JabRef
2. Go to General > Keyword and type
3. You'll see a list of unsorted keywords that make it hard to pick a particular one

![jabref4-1_selectors](https://user-images.githubusercontent.com/1692563/36835993-4707a054-1d38-11e8-9d26-a44ad67b47f8.png)

Alphabetical order would be great.

**Comment by stefan-kolb** [comment]
@JabRef/developers @Holzhauer This is auto-completion functionality? or is there any special functionality for managing keywords?

**Comment by stefan-kolb** [comment]
Ok this keyword manage dialog does not dunction as exptected. Shows different keywords for different selected entries sometimes non. adding does only partially work for me...

**Comment by jkrenz** [comment]
I just updated to a newer JabRef version on my new Mac (4.3.1). I think the problem still exists and it's horrible! I have over 100 passwords, so it's very time-consuming to add new entries to the database now. With the (very) old version I was using on my old mac it worked fine and I also found it much easier to add new keywords (you could do it from the entry).

**Comment by koppor** [comment]
Could you please try 5.0 alpha? We reworked the UI quite a bit; maybe, also
your issue is resolved?

jkrenz <notifications@github.com> schrieb am Do., 19. Sep. 2019, 15:54:

**Comment by AEgit** [comment]
JabRef 5.0-dev
Linux 5.0.0-29-generic amd64 
Java 11.0.4

Cannot confirm this issue for the current dev snap version. No selection menu at all is shown (indeed, I thought the autocompletion feature had been removed for the time, because it caused too many issues?).

**Comment by jkrenz** [comment]
Problem still persists in JabRef 5.0 for Mac. Keywords not in alphabetical order, neither when I want to add a new keyword ("Manage content selectors") nor when I want to choose a keyword for an entry. 

![Bildschirmfoto 2019-09-23 um 10 20 22](https://user-images.githubusercontent.com/37696037/65411159-723afe80-ddec-11e9-95fb-0d27696b8dff.png)
![Bildschirmfoto 2019-09-23 um 10 21 41](https://user-images.githubusercontent.com/37696037/65411166-77984900-ddec-11e9-9158-c8bc50dc415d.png)

**Comment by jkrenz** [comment]
Can anyone tell me which Jabref (for mac) version was the last one where the alphabetical order worked and where I can find it?

**Comment by BJaroszkowski** [comment]
I was not able to confirm the issue when selecting a keyword for an entry - the list seems to be in alphabetical order whether the autocomplete is on or off. It is true however, that the keywords were stored in Content Selectors and displayed in Content Selector Manager in the order in which they were added rather than the alphabetical one. I fixed that by simply sorting the list whenever new entry is added. Additionally, I wrote tests for both ContentSelectorSuggestionProvider and ContentSelectorDialogViewModel (as they did not exists) that check among other things that the keywords are in alphabetical order. Is it OK for me to create a pull request?
